example/spec/parallel/ingestGranule/IngestGranuleWithModificationSpec.js

```diff
@@ -0,0 +1,243 @@
+          return executionId === ingestGranuleRule1.payload.testExecutionId;
+        },
+        {
+          timestamp__from: ingestTime,
+          'originalPayload.testExecutionId': ingestGranuleRule1.payload.testExecutionId,
+        },
+        { timeout: 30 }
+      );
+      console.log('ingestGranuleExecution1Arn', ingestGranuleExecution1Arn);
+
+      // Wait for the execution to be completed
+      await getExecutionWithStatus({
+        prefix,
+        arn: ingestGranuleExecution1Arn,
+        status: 'completed',
+      });
+
+      // Wait for the granule to be fully ingested
+      await getGranuleWithStatus({ prefix, granuleId, status: 'completed' });
+
+      // Create a new collection
+      newCollection = await createCollection(
+        prefix,
+        {
+          duplicateHandling: 'version',
+          process: 'modis',
+        }
+      );
+
+      // Re-ingest the updated granule
+      ingestGranuleRule2 = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'IngestGranule',
+          collection: pick(newCollection, ['name', 'version']),
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-'),
+            granules: [
+              {
+                granuleId,
+                collectionId: constructCollectionId(newCollection.name, newCollection.version),
+                version: collection.version,
+                files: [
+                  {
+                    name: fileName,
+                    path: sourcePath,
+                  },
+                ],
+              },
+            ],
+          },
+        }
+      );
+
+      // Find the execution ARN
+      console.log('ingestGranuleRule2.payload.testExecutionId', ingestGranuleRule2.payload.testExecutionId);
+      ingestGranuleExecution2Arn = await findExecutionArn(
+        prefix,
+        (execution) => {
+          const executionId = get(execution, 'originalPayload.testExecutionId');
+          return executionId === ingestGranuleRule2.payload.testExecutionId;
+        },
+        {
+          timestamp__from: ingestTime,
+          'originalPayload.testExecutionId': ingestGranuleRule2.payload.testExecutionId,
+        },
+        { timeout: 30 }
+      );
+
+      console.log('ingestGranuleExecution2Arn', ingestGranuleExecution2Arn);
+
+      // Wait for the execution to be completed
+      await getExecutionWithStatus({
+        prefix,
+        arn: ingestGranuleExecution2Arn,
+        status: 'completed',
+      });
+    } catch (error) {
+      beforeAllError = error;
+      throw error;
+    }
+  });
+
+  it('does not modify the granule collectionId', async () => {
+    if (beforeAllError) fail(beforeAllError);
+    else {
+      const reingestedGranule = await getGranule({
+        prefix,
+        granuleId,
+      });
+      // Make sure that the collectionId is unmodified
+      expect(reingestedGranule.collectionId).toBe(collectionId);
```

Comment:
This is the most important assertion to check

Reply:
I don't know how we can tell from the test, but I do know through CloudWatch logs 🤔 That's not very satisfying though. And yeah, this is part of the reason why I am not sure how the user will know that their write granule failed because there is no error that propagates to the workflow level. Let me think about this